crowd_anki/importer/import_dialog.py

```diff
@@ -0,0 +1,126 @@
+
+ConfigEntry = namedtuple("ConfigEntry", ["config_name", "default_value"])
+
+
+class ImportJsonSetting:
+    import_message: str
+    suggest_tag_imported_cards: bool
+
+    class Properties(Enum):
+        IMPORT_MESSAGE = ConfigEntry("import_message", "")
+        SUGGEST_TAG_IMPORTED_CARDS = ConfigEntry("suggest_tag_imported_cards", False)
+
+    def __init__(self, settings_dict: dict):
+        for prop in self.Properties:
+            setattr(self, prop.value.config_name, settings_dict.get(prop.value.config_name, prop.value.default_value))
+
+
+class ImportConfig:
+    personal_fields: List[Tuple[str, str]] = []
+    add_tag_to_cards: List[str] = []
+
+    use_header: bool = True
+    use_note_models: bool = True
+    use_notes: bool = True
+    use_media: bool = True
+
+    ignore_deck_movement: bool = False
+
+    def __repr__(self):
+        return f"ImportConfig({self.personal_fields!r}, {self.add_tag_to_cards!r}, " \
+               f"{self.use_header!r}, {self.use_notes!r}, {self.use_note_models!r}, {self.use_media!r} " \
+               f"{self.ignore_deck_movement!r})"
+
+
+class ImportDialog(QDialog):
+    def __init__(self, deck_json,
+                 parent=None):
+        super().__init__(None)
+        self.parent = parent
+        self.form = ConfigUI()
+        self.form.setupUi(self)
+        self.userConfig = ConfigSettings.get_instance()
+        self.deck_json = deck_json
+        self.import_settings = ImportJsonSetting(self.deck_json.get(SETTINGS_FIELD_NAME, {}))
+        self.ui_initial_setup()
+
+        self.final_import_config: ImportConfig = None
+
+    def accept(self):
+        self.get_import_config()
+        super().accept()
+
+    def reject(self):
+        self.get_import_config()
+        super().reject()
+
+    def ui_initial_setup(self):
+        self.setup_personal_field_selection()
+        self.setup_misc()
+
+    def setup_personal_field_selection(self):
+        def is_field_personal(f):
+            if "personal_field" not in f:
+                return False
+            return f["personal_field"]
+
+        for model in self.deck_json["note_models"]:
+            heading_ui = QListWidgetItem(model["name"])
+            heading_ui.setFlags(Qt.ItemIsEnabled)
+            self.form.list_personal_fields.addItem(heading_ui)
+
+            for field in model["flds"]:
+                field_ui = QListWidgetItem(field["name"])
+                field_ui.setCheckState(Qt.Checked if is_field_personal(field) else Qt.Unchecked)
+                field_ui.setFlags(Qt.ItemIsEnabled | Qt.ItemIsUserCheckable)
+                self.form.list_personal_fields.addItem(field_ui)
+
+    def setup_misc(self):
+        self.form.import_message_textbox.setText(self.import_settings.import_message)
+
+        if self.import_settings.suggest_tag_imported_cards:
+            self.form.cb_tag_cards.setCheckState(Qt.Checked)
+            self.form.cb_tag_cards.setText("Tag Cards (Suggested by Deck Manager!)")
```

Comment:
what's deck manager? (I may guess that you're talking about `anki-dm` or some-such, but I don't want that to be a prerequisite knowledge for using CrowdAnki)

Reply:
Deck Owner?

Reply:
eh, I still like the "Curator" best, owner has different connotations for me :p. how about "Maintainer" ?